modules/command-engine/core/src/main/scala/surge/core/KafkaProducerActor.scala

```diff
@@ -161,24 +192,62 @@ class KafkaProducerActor(
     metrics: Metrics,
     aggregateName: String,
     val assignedPartition: TopicPartition,
-    override val signalBus: HealthSignalBusTrait)
+    override val signalBus: HealthSignalBusTrait,
+    trackerTimeout: FiniteDuration = TimeoutConfig.PublisherActor.trackerTimeout)
     extends HealthyComponent
     with HealthSignalBusAware {
   private implicit val executionContext: ExecutionContext = ExecutionContext.global
   private val log = LoggerFactory.getLogger(getClass)
+  private val inflight: mutable.Set[MessageTracker] = mutable.Set.empty
```

Comment:
I think the set of tracked messages wants to live in the actor for thread safety - outside of the actor there's no guarantee two things won't try to mutate this set at the same time and step on one another, but within the actor we're guaranteed to process one message at a time and avoid that race condition.

That might mean a few other things will need to move around as well - `KTablePersistenceSupport` will no longer have access to the tracker directly if it lives in the actor state, so it can just blindly retry and we can let the smarter producer actor sort out duplicates

Reply:
done